kmerexpr/exp_grad_solver.py

```diff
@@ -96,22 +110,26 @@ def exp_grad_solver(loss_grad,  x_0, lrs=None, tol=10**(-8.0), gtol = 10**(-8.0)
                 lrst = lrs[iter]
             # Update using prod exp function
             x_new = prod_exp_normalize(x, lrst*grad)
-        # x_new = softmax(np.log(x) +lrst*grad)  # 
+        # x_new = softmax(np.log(x) +lrst*grad)  #
         if np.isnan(x_new.sum()):
             print("iterates have a NaN a iteration ",iter, " existing and return previous iterate" )
             break
         x_av = momentum*x_av +(1-momentum)*x_new
         loss, grad_new  = loss_grad(x_new, Hessinv = Hessinv)
-        
+
         # Checking if method is stopping
         if norm(x_new - x, ord =1) <= tol:
             print("Exp_grad iterates are less than: " + str(tol), " apart. Stopping")
             break
-        if norm(grad_new -grad , ord =1)<= gtol: 
+        if norm(grad_new -grad , ord =1)<= gtol:
```

Comment:
[optional]

We should really be running through code formatters, but if not, I'd prefer spaces around operators and no spaces before commas in arg lists.  So

```python
if norm(grad_new - grad, ord = 1) <= gtol:
```